### PR TITLE
Move CAS operations in TokenService to sequence numbers

### PR DESCRIPTION
**Pull request by bleskes**
Relates #37872 
Relates #10708

### CODE REVIEW

**Comment by bleskes** [comment]
@elasticmachine run elasticsearch-ci/1 please.

**Comment by bleskes** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java [hunk 1]
```diff
@@ -744,13 +745,17 @@ private void innerRefresh(String tokenDocId, Authentication userAuth, ActionList
...
-                                        .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL)
-                                        .request();
-                                executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN, updateRequest,
+                                        .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL);
+                                if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_6_7_0)) {
+                                    updateRequest.setIfSeqNo(response.getSeqNo());
+                                    updateRequest.setIfPrimaryTerm(response.getPrimaryTerm());
+                                } else {
+                                    updateRequest.setVersion(response.getVersion());
+                                }
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java]: Small suggestion to make the condition tight proof (an assert could also work).
```
if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_6_7_0)
  && response.getSeqNo() != UNASSIGNED_SEQ_NO) {
    updateRequest.setIfSeqNo(response.getSeqNo());
    updateRequest.setIfPrimaryTerm(response.getPrimaryTerm());
} else {
    updateRequest.setVersion(response.getVersion());
}
```
There might be guarantees in other places in the code, I haven't looked for it thoroughly, you know better.

**Review by albertzaharovits — APPROVED** [review]
LGTM

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java (hunk 1 above)
Comment (bleskes) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java]: @albertzaharovits sorry,  I didn't see this version of the comment before merging. I'll add it in a follow up.

**Pull request merged**